technique/muon/zero_field_setup.py

```diff
@@ -0,0 +1,278 @@
+        Check if there is huge stray field
+        :return: True if there is huge stray field
+        """
+        measured_x, measured_y, measured_z = self.read_values()
+        if (abs(measured_x) < 4000 and abs(measured_y)
+                < 4000 and abs(measured_z) < 4000):
+            return False
+        return True
+
+    def get_single_corrected_field_value(self, field_direction):
+        """
+        get single corrected field value
+        :param field_direction: X, Y or Z
+        :return: corrected field value from the given direction
+        """
+        return g.get_pv(
+            "TE:NDLT1205:ZFMAGFLD_01:{}:CORRECTEDFIELD".format(field_direction))
+
+    def ask_user_to_continue(self):
+        """
+        Ask user if they want to continue further
+        :return:
+        """
+        reply = ""
+        while reply != "Y" and reply != "N":
+            reply = input("Would you like to continue? Y/N\n")
+        if reply == "N":
+            return "N"
+
+    def get_instrument_name(self):
+        """
+        get instrument name
+        :return: instrument name
+        """
+        return g.get_instrument()
+
+
+def update_offset():
+    """
+    updates the offset so that the corrected field reading is 0 for each axis
+    :return: None
+    """
+    x_raw_measurement = g.get_pv("TE:NDLT1205:ZFMAGFLD_01:DAQ:X:_RAW")
+    y_raw_measurement = g.get_pv("TE:NDLT1205:ZFMAGFLD_01:DAQ:Y:_RAW")
+    z_raw_measurement = g.get_pv("TE:NDLT1205:ZFMAGFLD_01:DAQ:Z:_RAW")
+
+    x_new_offset = x_raw_measurement - 0
+    y_new_offset = y_raw_measurement - 0
+    z_new_offset = z_raw_measurement - 0
+
+    g.set_pv("TE:NDLT1205:ZFMAGFLD_01:X:OFFSET", x_new_offset)
+    g.set_pv("TE:NDLT1205:ZFMAGFLD_01:Y:OFFSET", y_new_offset)
+    g.set_pv("TE:NDLT1205:ZFMAGFLD_01:Z:OFFSET", z_new_offset)
+
+
+def run_all():
+    """
+    Run all the functions in order for set up
+    :return:
+    """
+    while True:
+        print("***********************")
+        procedure = ZeroFieldSetupProcedure()
+        procedure.set_setpoints_to_zero(0, 0, 0)
+        # 2 second timer
+        if procedure.check_if_stray_field_exist():
+            print("Warning! Magnet Not in Range")
+            # pylint: disable=E0602
+
+        current_x, current_y, current_z, fields_x, fields_y, fields_z = procedure.get_correlated_current_and_fields(
+            plot=True)
+
+        calibration_coefficient_x, coefficient_of_determination_x = procedure.calculate_coefficient_and_r_squared(
+            current_x, fields_x)
+        calibration_coefficient_y, coefficient_of_determination_y = procedure.calculate_coefficient_and_r_squared(
+            current_y, fields_y)
+        calibration_coefficient_z, coefficient_of_determination_z = procedure.calculate_coefficient_and_r_squared(
+            current_z, fields_z)
+
+        print(
+            "Calibration coefficient for X axis is {unknown}".format(
+                unknown=calibration_coefficient_z))
```

Comment:
This should actually be printing `calibration_coefficient_z`. In general these strings can just be `"Calibration coefficient for X axis is {}".format(calibration_coefficient_z))`.